<launch>

  <arg name="world" default="ens32_world" />
  <arg name="map_name_1" default="ens1" />
  <arg name="map_file_1" default="$(find bwi_gazebo)/maps/ens-basement-2012-09-04-simulation-1.yaml"/>
  <arg name="map_name_2" default="ens2" />
  <arg name="map_file_2" default="$(find bwi_gazebo)/maps/ens-basement-2012-09-04-simulation-2.yaml"/>
  <arg name="person1_ns" default="person1"/>
  <arg name="person_id" value="1"/>

  <!-- Start the bwi map server with the ens split map specifications-->
  <node name="bwi_map_server" pkg="bwi_apps" type="bwi_map_server">
    <param name="maps/$(arg map_name_1)/yaml" value="$(arg map_file_1)" />
    <rosparam param="maps/$(arg map_name_1)/origin">[0, 0, 0, 0, 0, 0]</rosparam>
    <param name="maps/$(arg map_name_2)/yaml" value="$(arg map_file_2)" />
    <rosparam param="maps/$(arg map_name_2)/origin">[27.5, 0, 0, 0, 0, 0]</rosparam>
    <rosparam param="links">[{from_point: {level_id: ens1, point: [8, 0.5, 0]}, to_point: {level_id: ens2, point: [-9, 0.5, 0]}}]</rosparam>
  </node>

  <!-- Spawn the person controller -->
  <group ns="$(arg person1_ns)">
    <node name="person_controller" pkg="bwi_person_controller" type="bwi_person_controller" output="screen">
      <param name="person_id" value="$(arg person_id)"/>
    </node>
  </group>

  <!-- Aggregate person locations (i.e. the fake person location provider) -->
  <node name="fake_person_location_system" pkg="bwi_apps" type="person_location_aggregator" output="screen">
    <param name="nodes/$(arg person1_ns)" value="$(arg person1_ns)" />
  </node>

  <!-- Aggregate robot locations from tf tree -->
  <node name="robot_location_aggregator" pkg="bwi_apps" type="robot_location_aggregator">
    <param name="robot_list_file" value="$(find bwi_apps)/config/robots.yaml" />
  </node>

  <!-- Start the path planner -->
  <node name="path_provider" pkg="bwi_apps" type="path_provider" />

  <!-- Start BWI Navigator -->
  <node name="bwi_navigator" pkg="bwi_apps" type="bwi_navigator" output="screen">
    <param name="robot_list/ens1" value="segbot1" />
    <param name="robot_list/ens2" value="segbot2" />
  </node>

</launch>
